x-pack/platform/plugins/private/intercepts/public/prompter/service/intercept_dialog_service.tsx

```diff
@@ -60,43 +64,66 @@ export class InterceptDialogService {
       throw new Error('Notifications coordinator is not initialized');
     }
 
-    // leverage the notifications coordinator to ensure that we are not showing
-    // multiple intercepts at the same time, and that we are not showing
+    // leverages the notifications coordinator to ensure that we are not showing
     // intercepts when the user is interacting with other notifications
     // (e.g. toast notifications)
-    const intercept$ = this.notificationsCoordinator
+    this.interceptsStream$ = this.notificationsCoordinator
       .optInToCoordination(get$(), ({ locked }) => !locked)
       .pipe(
-        Rx.mergeMap((x) => x),
-        // since the backing store for product intercepts accepts all queued intercepts,
-        // we might receive a list of intercepts that should be attended to,
-        // hence we attempt to present them serially to the user, without interrupting other queued notification types
-        Rx.delayWhen(() =>
-          this.notificationsCoordinator!.lock$.pipe(
-            Rx.filter(({ controller }) => controller === InterceptDialogService.name)
-          )
-        )
+        Rx.scan((pendingInterceptsQueue, incomingIntercepts) => {
+          const pendingInterceptIds = pendingInterceptsQueue.map((intercept) => intercept.id);
+
+          const newIncomingIntercepts = incomingIntercepts.filter(
+            (intercept) => !pendingInterceptIds.includes(intercept.id)
+          );
+
+          // Here we only return the new incoming intercepts,
+          // because we intend to hold on to all previous intercept values till they can be emitted
+          // under the conditions we define
+          return newIncomingIntercepts;
+        }, [] as Intercept[]),
+        // don't emit on empty
+        Rx.filter((pendingInterceptsQueue) => Boolean(pendingInterceptsQueue.length)),
+        Rx.concatMap((pendingInterceptsQueue) => {
+          return Rx.from(pendingInterceptsQueue).pipe(
+            Rx.concatMap((item) =>
+              this.notificationsCoordinator!.lock$.pipe(
+                Rx.filter(
+                  (lockState) =>
+                    !lockState.locked || lockState.controller === InterceptDialogService.name
+                ),
+                Rx.take(1), // Wait until the lock is available, then proceed
+                Rx.tap((lockState) => {
+                  if (!lockState.locked) {
+                    this.notificationsCoordinator!.acquireLock();
+                  }
+                }),
```

Comment:
We acquire a lock here, because we might have an intercept that was queued but was not able to be displayed at the time it was queued, if on attempting to display the intercept the lock isn't owned we acquire one to ensure the intercept remains the only notification in the system during it's lifetime.